<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data input and pre-processing | QRAP: an R package and Shiny app for interactive RNA sequencing data analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.30 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data input and pre-processing | QRAP: an R package and Shiny app for interactive RNA sequencing data analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data input and pre-processing | QRAP: an R package and Shiny app for interactive RNA sequencing data analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Shixue Gou" />


<meta name="date" content="2022-12-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="data-quality-exploring.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="intro.html"><a href="intro.html#r-package"><i class="fa fa-check"></i><b>1.2.1</b> R package</a></li>
<li class="chapter" data-level="1.2.2" data-path="intro.html"><a href="intro.html#docker-image"><i class="fa fa-check"></i><b>1.2.2</b> Docker image</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#get-started"><i class="fa fa-check"></i><b>1.3</b> Get started</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="intro.html"><a href="intro.html#launch-the-qrap-application-in-r-or-rstudio"><i class="fa fa-check"></i><b>1.3.1</b> Launch the QRAP application in R or Rstudio</a></li>
<li class="chapter" data-level="1.3.2" data-path="intro.html"><a href="intro.html#launch-the-qrap-application-by-docker-image-in-shell"><i class="fa fa-check"></i><b>1.3.2</b> Launch the QRAP application by docker image in shell</a></li>
<li class="chapter" data-level="1.3.3" data-path="intro.html"><a href="intro.html#access-the-interactive-analysis-interface"><i class="fa fa-check"></i><b>1.3.3</b> Access the interactive analysis interface</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-input-and-pre-processing.html"><a href="data-input-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Data input and pre-processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-input-and-pre-processing.html"><a href="data-input-and-pre-processing.html#data-input"><i class="fa fa-check"></i><b>2.1</b> Data input</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="data-input-and-pre-processing.html"><a href="data-input-and-pre-processing.html#upload-local-file"><i class="fa fa-check"></i><b>2.1.1</b> Upload local file</a></li>
<li class="chapter" data-level="2.1.2" data-path="data-input-and-pre-processing.html"><a href="data-input-and-pre-processing.html#pull-down-geo-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Pull down GEO datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="data-input-and-pre-processing.html"><a href="data-input-and-pre-processing.html#data-pre-processing"><i class="fa fa-check"></i><b>2.2</b> Data pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-quality-exploring.html"><a href="data-quality-exploring.html"><i class="fa fa-check"></i><b>3</b> Data quality exploring</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-quality-exploring.html"><a href="data-quality-exploring.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.1</b> principal component analysis (PCA)</a></li>
<li class="chapter" data-level="3.2" data-path="data-quality-exploring.html"><a href="data-quality-exploring.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>3.2</b> hierarchical clustering heatmap</a></li>
<li class="chapter" data-level="3.3" data-path="data-quality-exploring.html"><a href="data-quality-exploring.html#sample-to-sample-heatmap"><i class="fa fa-check"></i><b>3.3</b> sample-to-sample heatmap</a></li>
<li class="chapter" data-level="3.4" data-path="data-quality-exploring.html"><a href="data-quality-exploring.html#sample-correlation-coefficient"><i class="fa fa-check"></i><b>3.4</b> sample correlation coefficient</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>4</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#extract-degs"><i class="fa fa-check"></i><b>4.1</b> Extract DEGs</a></li>
<li class="chapter" data-level="4.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#visualize-degs"><i class="fa fa-check"></i><b>4.2</b> Visualize DEGs</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="deg-expression-pattern-detection.html"><a href="deg-expression-pattern-detection.html"><i class="fa fa-check"></i><b>5</b> DEG expression pattern detection</a>
<ul>
<li class="chapter" data-level="5.1" data-path="deg-expression-pattern-detection.html"><a href="deg-expression-pattern-detection.html#detect-deg-expression-pattern"><i class="fa fa-check"></i><b>5.1</b> Detect DEG expression pattern</a></li>
<li class="chapter" data-level="5.2" data-path="deg-expression-pattern-detection.html"><a href="deg-expression-pattern-detection.html#visualize-deg-expression-pattern"><i class="fa fa-check"></i><b>5.2</b> Visualize DEG expression pattern</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html"><i class="fa fa-check"></i><b>6</b> weighted correlation network analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#data-preparation"><i class="fa fa-check"></i><b>6.1</b> Data preparation</a></li>
<li class="chapter" data-level="6.2" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#soft-threshold-detection"><i class="fa fa-check"></i><b>6.2</b> Soft threshold detection</a></li>
<li class="chapter" data-level="6.3" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#gene-module-detection"><i class="fa fa-check"></i><b>6.3</b> Gene module detection</a></li>
<li class="chapter" data-level="6.4" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#module-traits-relationship"><i class="fa fa-check"></i><b>6.4</b> Module-Traits relationship</a></li>
<li class="chapter" data-level="6.5" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#mm-vs.-gs-scatterplot"><i class="fa fa-check"></i><b>6.5</b> MM vs. GS scatterplot</a></li>
<li class="chapter" data-level="6.6" data-path="weighted-correlation-network-analysis.html"><a href="weighted-correlation-network-analysis.html#module-gene-expression-visualization"><i class="fa fa-check"></i><b>6.6</b> Module gene expression visualization</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="functional-enrichment-analysis.html"><a href="functional-enrichment-analysis.html"><i class="fa fa-check"></i><b>7</b> functional enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="functional-enrichment-analysis.html"><a href="functional-enrichment-analysis.html#gprofiler-api"><i class="fa fa-check"></i><b>7.1</b> Gprofiler API</a></li>
<li class="chapter" data-level="7.2" data-path="functional-enrichment-analysis.html"><a href="functional-enrichment-analysis.html#clusterprofiler"><i class="fa fa-check"></i><b>7.2</b> ClusterProfiler</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="functional-enrichment-analysis.html"><a href="functional-enrichment-analysis.html#ora"><i class="fa fa-check"></i><b>7.2.1</b> ORA</a></li>
<li class="chapter" data-level="7.2.2" data-path="functional-enrichment-analysis.html"><a href="functional-enrichment-analysis.html#gsea"><i class="fa fa-check"></i><b>7.2.2</b> GSEA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="gene-regulatory-network.html"><a href="gene-regulatory-network.html"><i class="fa fa-check"></i><b>8</b> Gene regulatory network</a>
<ul>
<li class="chapter" data-level="8.1" data-path="gene-regulatory-network.html"><a href="gene-regulatory-network.html#kegg-pathview"><i class="fa fa-check"></i><b>8.1</b> KEGG Pathview</a></li>
<li class="chapter" data-level="8.2" data-path="gene-regulatory-network.html"><a href="gene-regulatory-network.html#ppi-network"><i class="fa fa-check"></i><b>8.2</b> PPI network</a></li>
<li class="chapter" data-level="8.3" data-path="gene-regulatory-network.html"><a href="gene-regulatory-network.html#genie3-inffered-network"><i class="fa fa-check"></i><b>8.3</b> GENIE3 inffered network</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="summary-of-genes-and-functions.html"><a href="summary-of-genes-and-functions.html"><i class="fa fa-check"></i><b>9</b> Summary of genes and functions</a>
<ul>
<li class="chapter" data-level="9.1" data-path="summary-of-genes-and-functions.html"><a href="summary-of-genes-and-functions.html#summarize-genes"><i class="fa fa-check"></i><b>9.1</b> summarize genes</a></li>
<li class="chapter" data-level="9.2" data-path="summary-of-genes-and-functions.html"><a href="summary-of-genes-and-functions.html#summarize-functions"><i class="fa fa-check"></i><b>9.2</b> summarize functions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">QRAP: an R package and Shiny app for interactive RNA sequencing data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-input-and-pre-processing" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data input and pre-processing<a href="data-input-and-pre-processing.html#data-input-and-pre-processing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="data-input" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Data input<a href="data-input-and-pre-processing.html#data-input" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>QRAP can support to analyze publicly available and user-generated data. There are two action buttons, ‘Get Started Local’ and ‘Get Started GEO,’ can activate corresponding data upload and processing pipeline.</p>
<p><img src="images/start_button.jpeg" /></p>
<div id="upload-local-file" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Upload local file<a href="data-input-and-pre-processing.html#upload-local-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For user generated local files, click the ‘Get Started Local’ button to enter the data upload page.</p>
<p><strong><em>Parameters in this section:</em></strong><br />
- Choose input file: browser and upload the expression matrix file, accept .csv/.tsv/.tab/.txt format.<br />
- First row as header ? This means use the first row of the expression matrix as column names, often is samples names.<br />
- First column as rownames ? This means use the first column of the expression matrix as row names, often is gene names.<br />
</p>
<p><img src="images/upload_local_file.jpeg" /></p>
<p>After select the local files and set the parameters, click the ‘Upload’ button to preview the expression matrix.</p>
<p><img src="images/local_expression_preview.jpeg" /></p>
<p>Click ‘Example’ will upload the example dataset internal, transcriptomes of SARS-Cov-2 infected normal human bronchial epithelial cells (GSE147507).</p>
</div>
<div id="pull-down-geo-datasets" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Pull down GEO datasets<a href="data-input-and-pre-processing.html#pull-down-geo-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Gene Expression Omnibus (GEO) is a public repository that archives and freely distributes comprehensive sets of microarray, next-generation sequencing, and other forms of high-throughput functional genomic data submitted by the scientific community.</p>
<p>After you input the GEO Acession Number and active the ‘Download matrix’ button, We will download the value matrix tables within GDSxxx or supplementary files within GSExxx, these files will store in the working directory of the R project you created.
<img src="images/search_geo_dataset.jpeg" /></p>
<p>When the files download accomplished, there will show the download files name in the Parameter setting panel, and you should select file(s) that contain interested gene expression matrix and active the ‘Loading GEO’ button to preview the matrix.
<img src="images/geo_preview_single.jpeg" />
Please Note that if the files are in a tar archive format, such as htseq-count generated results, these files will contain Gene ID and Gene Expression Value of each sample, respectively. Therefore, you need to provid the column number of the Gene ID and Gene Expression Value to help merge the files to generate an analysis ready gene expression matrix.</p>
<ul>
<li>Step 1. Choose a single file, and leave the ‘First row as header’ and ‘First column as rownames’ unchecked, then click the button ‘Loading GEO’ to preview what the file contained.
<img src="images/geo_preview_multiple_setp1.jpeg" /></li>
<li>Step 2. Choose all or interested files, set the column number of geneID and readCounts, then click the button ‘Loading GEO’ to generate and preview the analysis ready expression matrix.
<img src="images/geo_preview_multiple_setp2.jpeg" /></li>
</ul>
</div>
</div>
<div id="data-pre-processing" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Data pre-processing<a href="data-input-and-pre-processing.html#data-pre-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>QRAP starts with a read-counts matrix or GEO accession number, and then filters out low expression genes under an given threshold. We then specify the organisms and filtered out low expression genes.
<img src="images/prefiltering.jpeg" /></p>
<div id="note-the-gene-types" class="section level6 unnumbered hasAnchor">
<h6>Note the Gene Types:<a href="data-input-and-pre-processing.html#note-the-gene-types" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<table>
<thead>
<tr class="header">
<th align="left">SYMBOL</th>
<th align="left">ENSEMBL</th>
<th align="left">ENTREZID</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GAPDH</td>
<td align="left">ENSG00000111640</td>
<td align="left">2597</td>
</tr>
<tr class="even">
<td align="left">TP53</td>
<td align="left">ENSG00000141510</td>
<td align="left">7157</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-quality-exploring.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["QRAP_book.pdf", "QRAP_book.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
